fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/FDBLuceneQueryTest.java

```diff
@@ -0,0 +1,309 @@
+        try (FDBRecordContext context = openContext()) {
+            openRecordStore(context);
+            final QueryComponent filter1 = new LuceneQueryComponent("civil blood makes civil hands unclean");
+            final QueryComponent filter2 = new LuceneQueryComponent("was king from 966 to 1016");
+            // Query for full records
+            RecordQuery query = RecordQuery.newBuilder()
+                    .setRecordType(TextIndexTestUtils.SIMPLE_DOC)
+                    .setFilter(Query.or(filter1, filter2))
+                    .build();
+            setDeferFetchAfterUnionAndIntersection(shouldDeferFetch);
+            RecordQueryPlan plan = planner.plan(query);
+            Matcher<RecordQueryPlan> matcher = union(
+                            filter(filter1, indexScan(allOf(indexScan("Complex$text_index"),
+                                    indexScanType(IndexScanType.BY_LUCENE),
+                                    bounds(hasTupleString("[[civil blood makes civil hands unclean],[civil blood makes civil hands unclean]]"))))),
+                            filter(filter2, indexScan(allOf(indexScan("Complex$text_index"),
+                                    indexScanType(IndexScanType.BY_LUCENE),
+                                    bounds(hasTupleString("[[was king from 966 to 1016],[was king from 966 to 1016]]"))))),
+                            equalTo(field("doc_id"))
+                    );
+            assertThat(plan, matcher);
+            List<Long> primaryKeys = recordStore.executeQuery(plan).map(FDBQueriedRecord::getPrimaryKey).map(t -> t.getLong(0)).asList().get();
+            assertEquals(ImmutableSet.of(1L, 2L, 4L), ImmutableSet.copyOf(primaryKeys));
+            if (shouldDeferFetch) {
+                assertLoadRecord(5, context);
+            } else {
+                assertLoadRecord(6, context);
+            }
+        }
+    }
+
+    @ParameterizedTest
+    @BooleanSource
+    public void delayFetchOnIntersectionOfLuceneScanWithFilters(boolean shouldDeferFetch) throws Exception {
+        try (FDBRecordContext context = openContext()) {
+            openRecordStore(context);
+            final QueryComponent filter1 = new LuceneQueryComponent("the continuance");
+            final QueryComponent filter2 = new LuceneQueryComponent("grudge");
+            // Query for full records
+            RecordQuery query = RecordQuery.newBuilder()
+                    .setRecordType(TextIndexTestUtils.SIMPLE_DOC)
+                    .setFilter(Query.and(filter1, filter2))
+                    .build();
+            setDeferFetchAfterUnionAndIntersection(shouldDeferFetch);
+            RecordQueryPlan plan = planner.plan(query);
+            Matcher<RecordQueryPlan> matcher = filter(filter2, indexScan(allOf(indexScanType(IndexScanType.BY_LUCENE),
+                    indexName("Complex$text_index"),
+                    bounds(hasTupleString("[[the continuance],[the continuance]]")))));
```

Comment:
We've lost `grudge`, haven't we?

Reply:
grudge is included in the filter2 LuceneQueryComponent.

Reply:
Right. But aside from the temporary state where that returns `true` for everything, is it in fact the future intention to be able to match a query string against a text string at runtime? I was assuming not so that all Lucene queries needed to go into the `search` argument.

Reply:
What do you mean by the search argument? I may be missing something about the use case. I assumed we wanted to be able to search distinct strings that appeared separately in the document.

Reply:
Yes, I agree that's what we want. Isn't that what `"the continuance" AND grudge` would do? That's why I thought maybe combining separate queries with simple string operations would work.

Reply:
So follow up thought. Do we want to support both functionality or should we cause an issue if we have two Lucene queries in the same request?

Reply:
If we transform it in the record layer then I guess it doesn't matter. I'll write extra testing for both cases.